drivers/at24cxxx/at24cxxx.c

```diff
@@ -97,35 +97,62 @@ int _read(const at24cxxx_t *dev, uint32_t pos, void *data, size_t len)
         }
         xtimer_usleep(AT24CXXX_POLL_DELAY_US);
     }
+
     DEBUG("[at24cxxx] i2c_read_regs(): %d; polls: %d\n", check, polls);
+
     return check;
 }
 
+static int _read_max(const at24cxxx_t *dev, uint32_t pos, void *data, size_t len)
+{
+#ifdef PERIPH_I2C_MAX_BYTES_PER_FRAME
+    uint8_t *data_p = (uint8_t*)data;
+    size_t bytes_read = 0;
+
+    while (len) {
+        size_t clen = MIN(len, PERIPH_I2C_MAX_BYTES_PER_FRAME);
+
+        if (_read(dev, pos, data_p, clen) == AT24CXXX_OK) {
+            bytes_read += clen;
+            len -= clen;
+            pos += clen;
+            data_p += clen;
+        }
+        else {
+            break;
```

Comment:
Not
```suggestion
            return -EIO;
```
?

Reply:
But it's hiding the error if one happened.
Also, what are you going to do with half a transaction? In case of the MTD layer, it will just retry because it expects `read_page` to return less than what was requested in case of  boundary condition.

With this, in case of an error, this would just return 0 and we will spin forever.

Reply:
So it should be the alternative that I proposed?

```
static int _mtd_at24cxxx_read_page(mtd_dev_t *mtd, void *dest, uint32_t page,
                                   uint32_t offset, uint32_t size)
{
    return at24cxxx_read(DEV(mtd), page * (mtd->page_size) + offset, dest, size) == AT24CXXX_OK ? (int)size : -EIO;
}
```

Reply:
> Also, what are you going to do with half a transaction?

MTD could request the second half in case something similar happens like only being able to read 256 bytes.

Reply:
BTW even when changing that, we can still input `len=0` and get a return of `0` and not `-EIO`, because we "successfully" read 0 bytes. If this is an issue we should forbid it I guess?

Reply:
`mtd_read_page()` does `while (count) {` so the `len = 0` case should never make it to the driver.